logrotate.c

```diff
@@ -2348,11 +2348,22 @@ static int rotateLogSet(const struct logInfo *log, int force)
         }
     }
 
-    if (log->rotateCount > 0)
-        message(MESS_DEBUG, "(%d rotations)\n", log->rotateCount);
-    else if (log->rotateCount == 0)
+    if (log->rotateCount == 0)
         message(MESS_DEBUG, "(no old logs will be kept)\n");
-
+    else {
+        if (log->rotateCount > 0)
+            message(MESS_DEBUG, "(%d rotations)", log->rotateCount);
+        else if (log->rotateCount == -1)
```

Comment:
This condition is redundant, `rotateCount` should never be lower than -1: https://github.com/logrotate/logrotate/blob/0cc114f42c03b7931bd3258405ea13005bfd058d/config.c#L1375